pkg/types/operartion_utils.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+func (e *Operation) IsAsyncResponse() bool {
+
+	if e.Context.IsAsyncDefinedByClient {
+		return e.Context.BrokerResponse.Async
```

Comment:
you don't need this property anymore (BrokerResponse.Async)
now you have indication if the user pass async or not
you can update the async to true (in the interceptor) in case of async response and !Context.IsAsyncDefinedByClient

and you can just return e.Context.Async